src/net.cpp

```diff
@@ -1819,6 +1819,14 @@ void CConnman::CreateNodeFromAcceptedSocket(std::unique_ptr<Sock>&& sock,
     }
 
     const bool inbound_onion = std::find(m_onion_binds.begin(), m_onion_binds.end(), addr_bind) != m_onion_binds.end();
+
+    // Do not accept inbound connections from peers already connected to us.
```

Comment:
Just because the connections come from the same IP does no mean it is the same peer, e.g. multiple peers could be connecting to you through the same Tor exit relay.

This does not seem like a bug fix and more like a behavior change. Our inbound eviction logic should handle stuff like this and it probably already does.

Reply:
> Inbound tor peers are exempted from this check.

We don't have logic for detecting if someone is using Tor as a proxy to connect to the IPv4 network. Our "inbound Tor" classification simply refers to someone connecting to our hidden service.

Another example would be multiple peers connecting to us from the same VPN provider.

Reply:
Yes, different peers can be connecting to us through the same NAT box, Tor exit node, or VPN server. Or it might be the same peer indeed. I think that the pros of assuming it is the same peer out-weight the pros of assuming it is not. However, I can see how this can be arguable and without having some solid numbers to back either argument maybe best to keep this unchanged.